mlflow/pyfunc/context.py

```diff
@@ -55,3 +55,16 @@ def get_prediction_context() -> Optional[Context]:
         The context for the current prediction request, or None if no context is set.
     """
     return _PREDICTION_REQUEST_CTX.get()
+
+
+@contextlib.contextmanager
+def potentially_set_prediction_context(context: Optional[Context]):
+    """
+    Only override the prediction context if it is provided. Otherwise, use the existing
+    prediction context or no prediction context at all.
+    """
+    if context:
+        with set_prediction_context(context):
+            yield
+    else:
+        yield
```

Comment:
/ai suggest names for this function. My fingers are not good at typing `potentially`.

Reply:
Do we need both `set_prediction_context` and `potentially_set_prediction_context`?

Reply:
Renamed to `maybe_set_prediction_context`.
`set_prediction_context` sets the context even if context is None, but `maybe_set_prediction_context` only sets context if it's not None. I want to avoid override existing context when this function is called.

Reply:
so `maybe_set_prediction_context` should be used when we want to avoid overriding the existing context?

Reply:
Is there a case where `set_prediction_context` should be used?

Reply:
So the difference only exists when `None` is passed, so sth like this;
```
with set_prediction_context(Context(...)):
    ...
    with set_prediction_context(None):   # If we use maybe_set_prediction_context, it won't override
        ...
```
Do we have this use case now....?

Reply:
Yes it's used a lot, and in databricks serving, we probably don't want to change that function